src/string/memmove.c

```diff
@@ -0,0 +1,23 @@
+#include <string.h>
+
+void *
+memmove(void *s1, const void *s2, size_t n)
+{
+	char *cs1;
+	const char *cs2;
+
+	cs1 = (char *)s1;
+	cs2 = (const char *)s2;
+	if (cs2 + n >= cs1)
```

Comment:
should be `cs1 >  cs2` also the case where cs1 == cs2 should be added (simply return s1)